modules/ModuleBuilder/parsers/views/PopupMetaDataParser.php

```diff
@@ -34,210 +34,237 @@
+    {
+        $viewdefs = $this->_viewdefs = array_change_key_case($viewdefs);
+        $viewdefs = $this->_viewdefs = $this->convertSearchToListDefs($viewdefs);
+
+        return $viewdefs;
+    }
+
+    function convertSearchToListDefs($defs)
+    {
+        $temp = array();
+        foreach ($defs as $key => $value) {
+            if (!is_array($value)) {
+                $temp[$value] = array('name' => $value);
+            } else {
+                $temp[$key] = $value;
+                if (isset($value['name']) && $value['name'] != $key) {
+                    $temp[$value['name']] = $value;
+                    unset($temp[$key]);
+                } else {
+                    if (!isset($value['name'])) {
+                        $temp[$key]['name'] = $key;
+                    }
+                }
+            }
+        }
+
+        return $temp;
+    }
+
+    function getOriginalViewDefs()
+    {
+        $defs = parent::getOriginalViewDefs();
+
+        return $this->convertSearchToListDefs($defs);
+    }
+
+    public function getSearchFields()
+    {
+        $searchFields = array();
+        foreach ($this->_viewdefs as $key => $def) {
+            if (isset($this->_fielddefs [$key])) {
+                $searchFields [$key] = self::_trimFieldDefs($this->_fielddefs [$key]);
+                if (!empty($def['label'])) {
+                    $searchFields [$key]['label'] = $def['label'];
+                }
+            } else {
+                $searchFields [$key] = $def;
+            }
+        }
+
+        return $searchFields;
     }
-	
-	function getOriginalViewDefs(){
-		$defs = parent::getOriginalViewDefs();
-		return $this->convertSearchToListDefs($defs);
-	}
-	
-	public function getSearchFields()
-	{
-		$searchFields = array ( ) ;
-        foreach ( $this->_viewdefs as $key => $def )
-        {
-            if (isset($this->_fielddefs [ $key ] )) {
-				$searchFields [ $key ] = self::_trimFieldDefs ( $this->_fielddefs [ $key ] ) ;
-				if (!empty($def['label']))
-				   $searchFields [ $key ]['label'] = $def['label'];
+
+    function handleSave($populate = true)
+    {
+        if (empty ($this->_packageName)) {
+            foreach (array(MB_CUSTOMMETADATALOCATION, MB_BASEMETADATALOCATION) as $value) {
+                $file = $this->implementation->getFileName(MB_POPUPLIST, $this->_moduleName, $value);
+                if (file_exists($file)) {
+                    break;
+                }
             }
-			else {
-				$searchFields [ $key ] = $def;
-			}
+            $writeFile = $this->implementation->getFileName(MB_POPUPLIST, $this->_moduleName);
+            if (!file_exists($writeFile)) {
+                mkdir_recursive(dirname($writeFile));
+            }
+        } else {
+            $writeFile = $file = $this->implementation->getFileName(MB_POPUPLIST, $this->_moduleName,
+                $this->_packageName);
+        }
+        $this->implementation->_history->append($file);
```

Comment:
undefined class field: _history, add it for class and/or create phpdoc for function